src/components/hero/interactive.tsx

```diff
@@ -0,0 +1,207 @@
+import React from "react";
+
+import { NotificationManager } from "react-notifications";
+import { serveFile } from "./utils";
+import {
+  InputFileSpecification,
+  ColumnSpecification,
+  BackButton,
+  SubmitButton,
+} from "./subcomponents";
+
+interface InteractiveTerminalProps {
+  onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+}
+
+interface InteractiveTerminalState {
+  column_id: string;
+  column_value: string;
+  column_sort: string;
+  column_kind: string;
+  data_file: string;
+  loading: boolean;
+}
+
+export default class InteractiveTerminal extends React.Component<
+  InteractiveTerminalProps,
+  InteractiveTerminalState
+> {
+  constructor(props) {
+    super(props);
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentWillMount() {
+    this.setState({
+      column_id: "",
+      column_value: "",
+      column_sort: "",
+      column_kind: "",
+      data_file: "",
+      loading: false,
+    });
+  }
+
+  handleChange(event) {
+    const target = event.target;
+    const value = target.value;
+    const name = target.name;
+
+    this.setState({
+      [name]: value,
+    });
+
+    if (target.type == "text") {
+      const canvas = document.createElement("canvas");
+      const context = canvas.getContext("2d");
+      context.font = window
+        .getComputedStyle(target, null)
+        .getPropertyValue("font");
+
+      const width = Math.max(context.measureText(value).width, 15);
+      target.style.width = Math.ceil(width) + "px";
+    }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const rawData = [...formData.entries()];
+    let params = new URLSearchParams(
+      rawData.filter((x) => x[1] != "" && x[0] != "data_file")
```

Comment:
```suggestion
      rawData.filter((x) => x[1] !== "" && x[0] !== "data_file")
```

But I would consider renaming `x` since it's kind of unreadable this way. What about `filter(([value_name_1, value_name_2]) => ...)` with better names for `value_name_1` and `value_name_2`.